ImputationPipeline/AggregatePRSResults.wdl

```diff
@@ -0,0 +1,235 @@
+version 1.0
+
+workflow AggregatePRSResults {
+  input {
+    Array[File] results
+    Array[File] target_pc_projections
+    File population_pc_projections
+    String population_name = "Reference Population"
+    File expected_control_results
+  }
+
+  call AggregateResults {
+    input:
+      results = results
+  }
+
+  call PlotPCA {
+    input:
+      batch_id = AggregateResults.batch_id,
+      population_name = population_name,
+      target_pc_projections = target_pc_projections,
+      population_pc_projections = population_pc_projections
+  }
+
+  call BuildHTMLReport {
+    input:
+      batch_id = AggregateResults.batch_id,
+      batch_all_results = AggregateResults.batch_all_results,
+      batch_control_results = AggregateResults.batch_control_results,
+      expected_control_results = expected_control_results,
+      batch_summarised_results = AggregateResults.batch_summarised_results,
+      score_distribution = AggregateResults.batch_score_distribution,
+      pc_plot = PlotPCA.pc_plot
+  }
+
+  output {
+    File batch_all_results = AggregateResults.batch_all_results
+    File batch_control_results = AggregateResults.batch_control_results
+    File batch_summarised_results = AggregateResults.batch_summarised_results
+    File score_distribution = AggregateResults.batch_score_distribution
+    File pc_polot = PlotPCA.pc_plot
+    File report = BuildHTMLReport.report
+  }
+}
+
+task AggregateResults {
+  input {
+    Array[File] results
+  }
+
+  command <<<
+    Rscript - <<- "EOF"
+    library(dplyr)
+    library(readr)
+    library(purrr)
+    library(tidyr)
+    library(magrittr)
+    library(ggplot2)
+
+    results <- c("~{sep='","' results}") %>% map(read_csv, col_types=cols(.default = 'c')) %>% reduce(bind_rows)
+    
+    num_batches <- lapply(results, function(row) {length(unique(row))})$batch_id
+
+    if (num_batches != 1) {
+      stop(paste0("There are ", num_batches, " batch IDs in the input tables, however, only 1 is expected."))
+    }
+
+    batch_id <- results$batch_id[1]
+
+    num_control_samples <- sum(results$is_control=="true")
+
+    if (num_batches != 1) {
+      stop(paste("There are ", num_control_samples, " control samples in the input tables, however, only 1 is expected."))
+    }
+
+    write_tsv(results, paste0(batch_id, "_all_results.tsv"))
+
+    write_tsv(results %>% filter(is_control == "true"), paste0(batch_id, "_control_results.tsv"))
```

Comment:
should be able to just do `filter(is_control)`.  "true" should be parsed as a logical by read_tsv.